src/service/resource.js

```diff
@@ -147,6 +147,9 @@ export class Resource {
     /** @private {!AmpElement|undefined|null} */
     this.owner_ = undefined;
 
+    /** @private {!AmpElement|undefined|null} */
+    this.ampAncestor_ = undefined;
```

Comment:
This needs to be discussed a bit more. But CSS changes seem rather clear to me. Can we merge them first and split this part into another PR? I'd like to get a bit deeper on this.